File: src/main/java/calculator/CalculatorController.java
package calculator;

import mvc.*;

/**
 * The CalculatorController class handles user input and coordinates between
 * the CalculatorModel and CalculatorView. It extends AbstractController
 * to provide MVC framework integration.
 */
public class CalculatorController extends AbstractController {
    
    /**
     * MANDATORY: Constructor that initializes the MVC components
     * Creates the model and view, and makes the view visible
     */
    public CalculatorController() {
        setModel(new CalculatorModel());
        setView(new CalculatorView((CalculatorModel)getModel(), this));
        ((JFrameView)getView()).setVisible(true);
    }
    
    /**
     * MANDATORY: Handle user operations from the view
     * 
     * @param option The operation string ("+", "-", "=", "Clr", or digit)
     */
    public void operation(String option) {
        if (option.equals(CalculatorView.MINUS)) {
            ((CalculatorModel)getModel()).subtract();
        } else if (option.equals(CalculatorView.PLUS)) {
            ((CalculatorModel)getModel()).add();
        } else if (option.equals(CalculatorView.CLEAR)) {
            ((CalculatorModel)getModel()).clear();
        } else if (option.equals(CalculatorView.EQUALS)) {
            ((CalculatorModel)getModel()).equals();
        } else {
            ((CalculatorModel)getModel()).store(Integer.parseInt(option));
        }
    }
}

File: src/main/python/calculator/calculator_controller.py
"""
The CalculatorController class handles user input and coordinates between
the CalculatorModel and CalculatorView. It extends AbstractController
to provide MVC framework integration.
"""
from mvc import AbstractController
from .calculator_model import CalculatorModel
from .calculator_view import CalculatorView


class CalculatorController(AbstractController):
    """
    MANDATORY: Constructor that initializes the MVC components
    Creates the model and view, and makes the view visible
    """
    def __init__(self):
        super().__init__()
        model = CalculatorModel()
        self.set_model(model)
        view = CalculatorView(model, self)
        self.set_view(view)
        view.mainloop()
    
    """
    MANDATORY: Handle user operations from the view
    
    Args:
        option: The operation string ("+", "-", "*", "=", "Clr", or digit)
    """
    def operation(self, option: str) -> None:
        model = self.get_model()
        if not isinstance(model, CalculatorModel):
            return
        
        if option == CalculatorView.MINUS:
            model.subtract()
        elif option == CalculatorView.PLUS:
            model.add()
        elif option == CalculatorView.MULTIPLY:
            model.multiply()
        elif option == CalculatorView.CLEAR:
            model.clear()
        elif option == CalculatorView.EQUALS:
            model.equals()
        else:
            try:
                value = int(option)
                model.store(value)
            except ValueError:
                pass  # Invalid input, ignore

